### PR TITLE
Add build target to Makefile

### LINKED ISSUE #2: Add 'build' and 'uninstall' make targets
Right now, it is not straightforward how to uninstall application. Having special target in Makefile called 'uninstall' would help a ton.

Another thing is current state of 'install' target: it combines building app (Python egg/wheel, translations, etc.) with installing them in proper locations. Installing in system locations requires root access and building app with root privileges is not good thing. So it would be nice to separate them.

### PR DESCRIPTION
**Pull request by scabala**
Closes #2 

Changelog:
 - added separate build target to Makefile.
 - modified install Makefile target to only copy files around

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,9 +21,31 @@ clean:
 	rm -rf */*/__pycache__
 	rm -rf */*/*/__pycache__
 
-install:
+.PHONY: build install
```
Comment (otsaloma) [inline Makefile:24]: There's a phony list at the end of the Makefile.

File: Makefile [hunk 2]
```diff
@@ -21,9 +21,31 @@ clean:
...
+
+build:
+	@echo "BUILDING PYTHON PACKAGE..."
+	python3 setup-partial.py build
+	@echo "BUILDING TRANSLATIONS..."
+	for LANG in `cat po/LINGUAS`; do \
+	echo $$LANG; \
+	mkdir -p build/$$LANG; \
+	msgfmt po/$$LANG.po -o build/$$LANG/catapult.mo; \
```
Comment (otsaloma) [inline Makefile:33]: I'd rather put these under a single directory, as it'd be cleaner once there's a large amount of languages, so maybe `build/mo/$$LANG.mo`?

File: Makefile [hunk 3]
```diff
@@ -21,9 +21,31 @@ clean:
...
+	@echo "BUILDING APPDATA FILE..."
+	mkdir -p build
+	msgfmt --xml -d po \
+	--template data/io.otsaloma.catapult.appdata.xml.in \
+	-o build/io.otsaloma.catapult.appdata.xml
+
+install: build
 	@echo "INSTALLING PYTHON PACKAGE..."
-	./setup-partial.py install $(if $(DESTDIR),--root=$(DESTDIR),) --prefix=$(PREFIX)
+	python3 setup-partial.py install $(if $(DESTDIR),--root=$(DESTDIR),) --prefix=$(PREFIX)
```
Comment (otsaloma) [inline Makefile:48]: Why this change?

**Review by otsaloma — CHANGES_REQUESTED** [review]

**Comment by otsaloma** [comment]
Also, two questions

1. What does this setup.py build/install split practically mean here? Because it seems to me that build only copies the files under the build directory and only once you run the install command, does it do some byte-compilation and egg creation. I'm just wondering if this setup.py build/install split makes some sense here?
2. Is it all right to have both setup.py and our Makefile commands write to the same build directory? I'm just checking that it doesn't disturb or confuse setup.py somehow?

File: Makefile (hunk 2 above)
Comment (scabala) [reply Makefile:33]: Sure, will do.

File: Makefile (hunk 3 above)
Comment (scabala) [reply Makefile:48]: Must have been some left-over from previous version.

**Comment by scabala** [comment]
1. Yes, indeed it does only copy files around. However, I would like to keep it separate. I don't know much about DEB side of the Force but in RPM land, I've seen spec files only with build and install calls to setup.py sepatated.
2. AFAIK, it does not. It has the benefit of having all build files in the same directory.

**Comment by otsaloma** [comment]
Thanks, I think this looks good now, merging.

**Review by otsaloma — APPROVED** [review]
